Source/WebGPU/WGSL/ConstantFunctions.h

```diff
@@ -151,7 +151,7 @@ static ConstantValue zeroValue(const Type* type)
             RELEASE_ASSERT_NOT_REACHED();
         },
         [&](const Types::Bottom&) -> ConstantValue {
-            RELEASE_ASSERT_NOT_REACHED();
+            return 0;
```

Comment:
When we infer bottom for a given expression it's an indication that there was an error computing the type for that expression, so it's not safe to generate constants for it.

If a bottom value made it this far, it's because we failed to handle it earlier on. From looking at the test case, the problem is a struct that has a field with an invalid type. Currently we set the type of the field to bottom, but we create (what appears to be) a valid struct type.

I think allowing bottom types in constant values is a slippery slope, because now the value does not match the type (since bottom by definition can't have a value). The current approach is more conservative: if a struct contains any bottom types, it's type should also be bottom. We already do that for other invalid field types in `TypeChecker::visit(AST::Structure&)`, but apparently I failed to handle the case where `memberType` itself is bottom.